### PR TITLE
HDDS-9208. Add queue limit in ReplicationServer.

### PR DESCRIPTION
**Pull request by z-bb**
## What changes were proposed in this pull request?

Add queue limit in ReplicationServer

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9208

### CODE REVIEW

**Comment by ChenSammi** [comment]
@z-bb , the patch looks good. Wait for the CI result.

**Comment by ChenSammi** [comment]
@z-bb , can you do a master rebase? The failed UT looks like be fixed by master branch.

**Comment by z-bb** [comment]
> @z-bb , can you do a master rebase? The failed UT looks like be fixed by master branch.

okey, I will do it

**Comment by ChenSammi** [comment]
@z-bb , do you think the failed "EC-recovery :: Wait for replication to succeed" is relevant?

**Comment by z-bb** [comment]
> @z-bb , do you think the failed "EC-recovery :: Wait for replication to succeed" is relevant?

@ChenSammi I'm not sure if it's related. I'm going to look into this, see the log is waiting for the container to replicate for more than 5 minutes

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java [hunk 1]
```diff
@@ -180,6 +185,18 @@ public static final class ReplicationConfig {
     )
     private int replicationMaxStreams = REPLICATION_MAX_STREAMS_DEFAULT;
 
+    /**
+     * The maximum of replication request queue length.
+     */
+    @Config(key = QUEUE_LIMIT,
+        type = ConfigType.INT,
+        defaultValue = "4096",
```
[the remark this answers is not in the document]
Comment (z-bb) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java:193]: @sumitagrawl, Do you think this has something to do with the failure of ci

[the remark this answers is not in the document]
Comment (sumitagrawl) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java:193]: @z-bb not related to CI failure, I mean what should be appropriate default value for nettyServer handling upload/download container ... as this is given to NettyServer

**Comment by ChenSammi** [comment]
@z-bb , would you do a master rebase again? There is one EC acceptance test #5260 fixed in master.

**Comment by z-bb** [comment]
> @z-bb , would you do a master rebase again? There is one EC acceptance test #5260 fixed in master.

okey, I have already rebased.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChenSammi) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java:193]: I guess 4096 this value is picked for the moment based on experience, @z-bb , right? 
And Guangbao, do you remember the the biggest queue length you have ever seen before without the queue limitation imposed?

[the remark this answers is not in the document]
Comment (z-bb) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java:193]: > I guess 4096 this value is picked for the moment based on experience, @z-bb , right? And Guangbao, do you remember the the biggest queue length you have ever seen before without the queue limitation imposed?

Yes, based on experience, we plan to add a monitor for the queue length. I will confirm this value later.

[the remark this answers is not in the document]
Comment (ChenSammi) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java:193]: @z-bb , I see.  We can revisit the default value of this "queue.limit" property once you got more info from the monitor history.

**Review by ChenSammi — APPROVED** [review]
Thanks @z-bb .

**Pull request merged**